cmd/chartsvc/main.go

```diff
@@ -47,6 +47,7 @@ func setupRoutes() http.Handler {
 	apiv1.Methods("GET").Path("/charts/{repo}/{chartName}").Handler(WithParams(getChart))
 	apiv1.Methods("GET").Path("/charts/{repo}/{chartName}/versions").Handler(WithParams(listChartVersions))
 	apiv1.Methods("GET").Path("/charts/{repo}/{chartName}/versions/{version}").Handler(WithParams(getChartVersion))
+	apiv1.Methods("POST").Path("/charts/resolve-repo").HandlerFunc(resolveRepo)
```

Comment:
I am not sure that POST is the correct verb for this endpoint. To me, POST is associated with a Non idempotent action (creation usually). In this case we have an idempotent action that returns a resource so it makes sense to me to use GET instead.

That said, I am assuming that you used POST to send the data in the body. I am wondering if we should send the manifest as base64encoded via query params instead.

`/charts/resolve-repo?manifest=[base64Manifest]`

Note that using Get with query params help with server side caching.

Reply:
I am not either convinced with the name of the endpoint either. This endpoint will return the latest versions of the chart in every single repo that it belongs to no?

Reply:
+1 I think we should try to use GET if we can. We'd need to check if there are any restrictions here on URL length though.

Reply:
at the end I've changed it because `/charts/resolve-repo/{chartName}` conflicts with `/charts/{repo}/{chartName}`. I think a neat solution would be `GET /repos/resolve/{chartName}?version={version}&appversion={appversion}`